swift_package/BrainFlow/Sources/BoardShim.swift

```diff
@@ -0,0 +1,606 @@
+        var len: Int32 = 0
+        var channels = [Int32](repeating: 0, count: 512)
+        let errorCode = get_ppg_channels (boardId.rawValue, preset.rawValue, &channels, &len)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return Array(channels[0..<Int(len)])
+    }
+
+    /**
+     * get row indices in returned by get_board_data() 2d array which contain accel
+     * data
+     */
+    static func getAccelChannels (_ boardId: BoardIds, _ preset: BrainFlowPresets = .AUXILIARY_PRESET) throws -> [Int32] {
+        var len: Int32 = 0
+        var channels = [Int32](repeating: 0, count: 512)
+        let errorCode = get_accel_channels (boardId.rawValue, preset.rawValue, &channels, &len)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return Array(channels[0..<Int(len)])
+    }
+
+    /**
+     * get row indices in returned by get_board_data() 2d array which contain analog
+     * data
+     */
+    static func getAnalogChannels (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> [Int32] {
+        var len: Int32 = 0
+        var channels = [Int32](repeating: 0, count: 512)
+        let errorCode = get_analog_channels (boardId.rawValue, preset.rawValue, &channels, &len)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return Array(channels[0..<Int(len)])
+    }
+
+    /**
+      * get row indices in returned by get_board_data() 2d array which contain gyro
+      * data
+      */
+    static func getGyroChannels (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> [Int32] {
+        var len: Int32 = 0
+        var channels = [Int32](repeating: 0, count: 512)
+        let errorCode = get_gyro_channels (boardId.rawValue, preset.rawValue, &channels, &len)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return Array(channels[0..<Int(len)])
+    }
+
+    /**
+     * get row indices in returned by get_board_data() 2d array which contain other
+     * data
+     */
+    static func getOtherChannels (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> [Int32] {
+        var len: Int32 = 0
+        var channels = [Int32](repeating: 0, count: 512)
+        let errorCode = get_other_channels (boardId.rawValue, preset.rawValue, &channels, &len)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return Array(channels[0..<Int(len)])
+    }
+
+    /*
+     If the board is streaming or playback, then get the master board ID from params.other_info.
+     Otherwise return the board ID itself.
+     */
+    static func getMasterBoardID(boardId: BoardIds, params: BrainFlowInputParams) throws -> BoardIds {
+        guard ((boardId == BoardIds.STREAMING_BOARD) || (boardId == BoardIds.PLAYBACK_FILE_BOARD)) else {
+            return boardId
+        }
+        if let otherInfoInt = Int32(params.other_info) {
```

Comment:
its not in other_info anymore, a special int filed was added

Reply:
I'm not sure which attribute you are referring to. As a guideline I'm looking at classbrainflow.board_shim.BrainFlowInputParams in :

https://brainflow.readthedocs.io/en/stable/UserAPI.html#python-api-reference

Reply:
Seems like docstring is missed https://github.com/brainflow-dev/brainflow/blob/master/python_package%2Fbrainflow%2Fboard_shim.py#L144 you also need to copy logic related to it from python

Reply:
Updated.